### DISCUSSION TITLE
How to deal with the long-tailed problem in dataset,like ava data?

### DISCUSSION THREAD

**Discussion by ZJU-lishuang** [open-mmlab/mmaction2#750] [Q&A]
How to deal with the long-tailed problem in dataset?
Is it useful to simply increase the amount of data in the tail?

**Comment by innerlee** [comment]
For imbalanced data, you can 

* change the sampling strategy, so that smaller classes are sampled more frequently, or
* change the sample weight in computing loss, or
* training specific models for the long-tailed subset

Which one is best is unknown

**Reply by ZJU-lishuang** [reply]
I will jsut try it.

**Reply by innerlee** [reply]
My recommendation is chaning the sampling. You can compute the median size of all classes, and in each epoch, randomly resample each class to have 2*median size samples (either subsampling or repeating, depending on the size of the class).

**Reply by irvingzhang0512** [reply]
WeightedRandomSampler is helpful. For now, WeightedRandomSampler is supported for single-gpu training.
Distributed WeightedRandomSampler is not supported in official pytorch yet.
